Neos.ContentRepository.TestSuite/Classes/Behavior/Features/Bootstrap/ProjectedNodeTrait.php

```diff
@@ -549,17 +563,17 @@ public function iExpectThisNodeToNotBeReferenced(): void
     public function iExpectThisNodeToBeTheChildOfNode(string $serializedParentNodeDiscriminator): void
     {
         $expectedParentDiscriminator = NodeDiscriminator::fromShorthand($serializedParentNodeDiscriminator);
-        $this->assertOnCurrentNode(function (Node $currentNode) use ($expectedParentDiscriminator) {
+        $this->assertOnCurrentContentStreamAwareNode(function (ContentStreamAwareNode $currentNode) use ($expectedParentDiscriminator) {
             $subgraph = $this->getCurrentSubgraph();
 
             $parent = $subgraph->findParentNode($currentNode->aggregateId);
             Assert::assertInstanceOf(Node::class, $parent, 'Parent not found.');
-            $actualParentDiscriminator = NodeDiscriminator::fromNode($parent);
+            $actualParentDiscriminator = NodeDiscriminator::fromNode($currentNode->builder()->buildNode($parent));
```

Comment:
alternative: ... if we dislike the god object

```suggestion
            $actualParentDiscriminator = NodeDiscriminator::fromNode(ContentStreamAwareNodeBuilder::fromNode($currentNode)->buildNode($parent));
```